src/components/WalletModal/index.tsx

```diff
@@ -211,13 +215,14 @@ export default function WalletModal({
   useEffect(() => {
     if (account && account !== lastActiveWalletAddress) {
       const walletType = getConnectionName(getConnection(connector).type)
+      const peerWalletName = provider ? getWalletMeta(provider)?.name : undefined
```

Comment:
I think you should use the agent instead of the name, as it will disambiguate how the wallet is being used (eg it will include things like if a user connected via qr, or connected via WC or Injected). This was the intention of making the more verbose `agent` available ([src](https://github.com/Uniswap/conedison/blob/main/src/provider/meta.ts#L32)) along with the `name`, which is intended for UI.

Reply:
Could you update the PR description to match this?